### PR TITLE
add plugin and fix bugs

### PR DESCRIPTION
**Pull request by VictorDu**

### CODE REVIEW

File: templates/default/20-cassandra.conf.erb [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+      <Value>
+        Type "counter"
+        InstancePrefix "cassandra.Compaction.TotalCompactionsCompleted.Count"
+        Attribute "Count"
+      </Value>
+    </MBean>
+
+    <Connection>
+      ServiceURL "<%= @serviceurl %>"
+      Host "<%= @hostname %>"
```
Comment (tedoc2000) [inline templates/default/20-cassandra.conf.erb]: add [hostHasService=cassandra] to Host "<%= @hostname %>" -> Host "<%= @hostname %>[hostHasService=cassandra]"

File: templates/default/20-kafka_82.conf.erb [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                Type "gauge"
+                Table false
+                Attribute "99thPercentile"
+                InstancePrefix "kafka.fetchproducetotal-time.99th"
+            </Value>
+        </MBean>
+
+        <Connection>
+            ServiceURL "<%= @serviceurl %>"
+            Host "<%= @hostname %>"
```
Comment (tedoc2000) [inline templates/default/20-kafka_82.conf.erb]: Host "<%= @hostname %>[hostHasService=kafka]"

File: templates/default/abc.py.erb [hunk 3]
```diff
@@ -0,0 +1,574 @@
+#! /usr/bin/python
```
Comment (tedoc2000) [inline templates/default/abc.py.erb]: This should be elasticsearch.py.erb?

File: README.md [hunk 4]
```diff
@@ -1,75 +1,133 @@
...
+
+### Using bootstrap ###
+
+After supplying attributes, use `knife bootstrap` to apply the recipes to Chef clients. For example, the following command applies the `chef_install_configure_collectd` recipe and then installs the Apache collectd plugin: 
+
+```knife bootstrap $ip --ssh-user $username --node-name $nodename --run-list 'recipe[chef_install_configure_collectd], recipe[chef_install_configure_collectd::apache]'```
+
+# License & Author #
 
 Author:: Wentao Du (<wentao@signalfx.com>)
```
Comment (mdubbyap) [inline README.md]: We should use support+chef@signalfx.com not your own name for public stuff

**Comment by mdubbyap** [comment]
Also, squash the commits.

File: recipes/config-collectd.rb [hunk 5]
```diff
@@ -1,15 +1,48 @@
...
+# Recipe:: config-collectd 
+#
+# Function:
+# This recipe can configure the basic Information for collectd
+#
+# Copyright (c) 2015 SignalFx, Inc, All Rights Reserved.
+
+require File.expand_path("../helper.rb", __FILE__)
+#
+# This fucntion is to get collectd configure file path. For centos and amazon,
```
Comment (tedoc2000) [inline recipes/config-collectd.rb]: spelling: function

File: recipes/config-write_http.rb [hunk 6]
```diff
@@ -1,49 +1,28 @@
...
-  source "10-write_http-plugin.conf.erb"
+# Recipe:: config-collectd
+#
+# Function:
+# This recipe can configure the write_http plugin to send metrics to signalfx
+#
+# Copyright (c) 2015 SignalFx, Inc, All Rights Reserved.
+
+#
+# This fucntion is to get collectd configure file path. For centos and amazon,
```
Comment (tedoc2000) [inline recipes/config-write_http.rb]: function

File: recipes/helper.rb [hunk 7]
```diff
@@ -0,0 +1,120 @@
+# down lood remote file
```
Comment (tedoc2000) [inline recipes/helper.rb]: #download

File: recipes/helper.rb [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  directory folder_path do
+    mode folder_mode
+    action :create
+    recursive true
+  end
+end
+
+# get aws instance unique id
+
+def getAWSInfor
```
Comment (tedoc2000) [inline recipes/helper.rb]: getAWSInfo ?

File: recipes/helper.rb [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+  begin
+    Timeout::timeout(10) do
+      aws_metadata = open('http://169.254.169.254/2014-11-05/dynamic/instance-identity/document'){ |io| data = io.read }
+      aws_JSON_Information = JSON.parse(aws_metadata)
+      return "#{aws_JSON_Information['instanceId']}_#{aws_JSON_Information['region']}_#{aws_JSON_Information['accountId']}"
+  end
+  rescue
+    Chef::Log.warn('Unable to get AWS instance ID, Timeout due to reading')
```
Comment (tedoc2000) [inline recipes/helper.rb]: Timeout while reading

File: recipes/install-collectd.rb [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+#
+
+def install_in_redhat(os, version)
+  install_repo_rpms(os, version)
+  install_package 'collectd'
+  install_package 'collectd-disk'
+end
+
+#
+# This fucntion is to update the ubuntu. It depends on the apt cookbook.
```
Comment (tedoc2000) [inline recipes/install-collectd.rb]: function

**Pull request merged**